src/octree/batch_iterator.rs

```diff
@@ -111,28 +111,38 @@ where
             position: self.position.split_off(0),
             layers,
         };
+        //println!("internal callback sending last batch...");
         (self.func)(point_data)
     }
 
-    fn push_point_and_callback(&mut self, point: Point) -> Result<()> {
-        self.push_point(point);
-        if self.position.len() == self.position.capacity() {
-            return self.callback();
+    fn push_point_and_callback<F2>(
+        &mut self,
+        point_iterator: FilteredPointsIterator<F2>,
+    ) -> Result<()>
+    where
+        F2: Fn(&Point) -> bool,
```

Comment:
Nit: Couldn't `F2` be `F` as well?  Or perhaps `Filter`?

Reply:
Filter sounds good!